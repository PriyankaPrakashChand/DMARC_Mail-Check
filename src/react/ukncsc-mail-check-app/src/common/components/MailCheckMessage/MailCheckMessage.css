.MailCheckMessage--container {
  border-width: 0px 0px 0px 8px;
  border-style: solid;
}
