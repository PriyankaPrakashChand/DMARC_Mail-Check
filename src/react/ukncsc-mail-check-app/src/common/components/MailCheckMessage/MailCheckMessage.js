import React from 'react';
import MarkDown from 'react-markdown';
import PropTypes from 'prop-types';
import { Message } from 'semantic-ui-react';
import { removeMailtoHyperlinksRenderer as link } from 'common/helpers/markdown';

import './MailCheckMessage.css';

/**
 * This component renders markdown from the API in GitHub Flavored Markdown: https://github.github.com/gfm/
 */

const MailCheckMessage = ({ children, markdown, fluid, ...props }) => (
  <Message
    {...props}
    style={{ margin: '0.2em 0em', maxWidth: fluid ? null : 600 }}
    className="MailCheckMessage--container"
  >
    {markdown ? <MarkDown source={children} renderers={{ link }} /> : children}
  </Message>
);

MailCheckMessage.defaultProps = {
  error: false,
  warning: false,
  success: false,
  fluid: false,
  info: false,
  markdown: false,
  size: null,
};

MailCheckMessage.propTypes = {
  error: PropTypes.bool,
  warning: PropTypes.bool,
  success: PropTypes.bool,
  fluid: PropTypes.bool,
  info: PropTypes.bool,
  markdown: PropTypes.bool,
  size: PropTypes.string,
};

export default MailCheckMessage;
