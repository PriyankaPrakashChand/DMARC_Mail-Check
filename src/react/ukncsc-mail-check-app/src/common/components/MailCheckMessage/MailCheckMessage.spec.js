import React from 'react';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';

import MailCheckMessage from './MailCheckMessage';

describe('MailCheckMessage', () => {
  let container;

  describe('when provided with no props', () => {
    beforeEach(() => {
      ({ container } = render(
        <MailCheckMessage>
          <p>security rulez</p>
        </MailCheckMessage>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when provided with the markdown flag', () => {
    beforeEach(() => {
      ({ container } = render(
        <MailCheckMessage markdown>
          {`- list item\n- another list item`}
        </MailCheckMessage>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('render the markdown as html', () =>
      expect(container.getElementsByTagName('li')).toHaveLength(2));
  });

  describe('when provided with the markdown flag and a mailto address', () => {
    beforeEach(() => {
      ({ container } = render(
        <MailCheckMessage markdown>mailto:me@ncsc.gov.uk</MailCheckMessage>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should not render a hyperlink', () =>
      expect(container.getElementsByTagName('a')).toHaveLength(0));

    test('it should contain the full mailto text', () =>
      expect(container.getElementsByTagName('p')[0]).toHaveTextContent(
        'mailto:me@ncsc.gov.uk'
      ));
  });

  describe('when provided with the markdown flag and a link', () => {
    beforeEach(() => {
      ({ container } = render(
        <MailCheckMessage markdown>https://ncsc.gov.uk</MailCheckMessage>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should render a hyperlink', () =>
      expect(container.getElementsByTagName('a')[0]).toHaveTextContent(
        'https://ncsc.gov.uk'
      ));
  });
});
