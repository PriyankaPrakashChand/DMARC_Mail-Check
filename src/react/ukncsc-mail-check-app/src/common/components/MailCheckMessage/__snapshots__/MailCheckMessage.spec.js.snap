// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailCheckMessage when provided with no props it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui message MailCheckMessage--container"
  >
    <p>
      security rulez
    </p>
  </div>
</div>
`;

exports[`MailCheckMessage when provided with the markdown flag and a link it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui message MailCheckMessage--container"
  >
    <div>
      <p>
        <a
          href="https://ncsc.gov.uk"
        >
          https://ncsc.gov.uk
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`MailCheckMessage when provided with the markdown flag and a mailto address it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui message MailCheckMessage--container"
  >
    <div>
      <p>
        mailto:me@ncsc.gov.uk
      </p>
    </div>
  </div>
</div>
`;

exports[`MailCheckMessage when provided with the markdown flag it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui message MailCheckMessage--container"
  >
    <div>
      <ul>
        <li>
          list item
        </li>
        <li>
          another list item
        </li>
      </ul>
    </div>
  </div>
</div>
`;
